Source/ACE.Entity/BaseAceObject.cs

```diff
@@ -107,66 +155,181 @@ public class BaseAceObject
+            }
+            set
+            {
+                var listItem = AceObjectPropertiesInt.Find(x => x.IntPropertyId == (uint)PropertyInt.HookType);
+                if (value != null)
+                    AceObjectPropertiesInt.Find(x => x.IntPropertyId == (uint)PropertyInt.HookType).PropertyValue = (uint)value;
+                else
+                    if (listItem != null)
+                    AceObjectPropertiesInt.Remove(listItem);
+            }
+        }
 
         /// <summary>
         /// TODO: convert to enum
         /// </summary>
-        [DbField("hookItemTypes", (int)MySqlDbType.UInt32)]
-        public uint HookItemTypes { get; set; }
+        public uint? HookItemTypes
+        {
+            get
+            {
+                return AceObjectPropertiesInt.Find(x => x.IntPropertyId == (uint)PropertyInt.HookItemType)?.PropertyValue;
+            }
+            set
+            {
+                var listItem = AceObjectPropertiesInt.Find(x => x.IntPropertyId == (uint)PropertyInt.HookItemType);
+                if (value != null)
+                    AceObjectPropertiesInt.Find(x => x.IntPropertyId == (uint)PropertyInt.HookItemType).PropertyValue = (uint)value;
+                else
+                    if (listItem != null)
+                    AceObjectPropertiesInt.Remove(listItem);
+            }
+        }
 
-        [DbField("itemsCapacity", (int)MySqlDbType.UByte)]
-        public byte ItemsCapacity { get; set; }
+        public byte? ItemsCapacity
+        {
+            get
+            {
+                return (byte?)AceObjectPropertiesInt.Find(x => x.IntPropertyId == (uint)PropertyInt.ItemsCapacity)?.PropertyValue;
+            }
+            set
+            {
+                var listItem = AceObjectPropertiesInt.Find(x => x.IntPropertyId == (uint)PropertyInt.ItemsCapacity);
+                if (value != null)
+                    AceObjectPropertiesInt.Find(x => x.IntPropertyId == (uint)PropertyInt.ItemsCapacity).PropertyValue = (byte)value;
+                else
+                    if (listItem != null)
+                    AceObjectPropertiesInt.Remove(listItem);
+            }
+        }
 
         /// <summary>
         /// TODO: convert to enum
         /// </summary>
-        [DbField("materialType", (int)MySqlDbType.UByte)]
-        public byte MaterialType { get; set; }
+        public byte? MaterialType
+        {
+            get
+            {
+                return (byte?)AceObjectPropertiesInt.Find(x => x.IntPropertyId == (uint)PropertyInt.MaterialType)?.PropertyValue;
+            }
+            set
+            {
+                var listItem = AceObjectPropertiesInt.Find(x => x.IntPropertyId == (uint)PropertyInt.MaterialType);
+                if (value != null)
+                    AceObjectPropertiesInt.Find(x => x.IntPropertyId == (uint)PropertyInt.MaterialType).PropertyValue = (byte)value;
+                else
+                    if (listItem != null)
```

Comment:
when the "else" is more than 1 line, as in this case, you need to use "{" "}" brackets.  that, and the "if" path needs indentation.  Pro tip: Ctrl+K, Ctrl+D to autoformat an entire file (don't do this in aligned enums tho)